main/autogen/autoloader.cc

```diff
@@ -210,7 +224,19 @@ string DefTree::renderAutoloadSrc(const core::GlobalState &gs, const AutoloaderC
             predeclare(gs, fullName, buf);
         }
 
-        if (!children.empty()) {
+        if (pkgName.exists()) {
+            ENFORCE(!gs.packageDB().empty());
+            const string_view shortName = pkgName.shortName(gs);
+            const string_view mungedName = shortName.substr(0, shortName.size() - core::PACKAGE_SUFFIX.size());
+
+            auto &pkg = gs.packageDB().getPackageInfo(pkgName);
+
+            // First path prefix is guaranteed to be the directory location of the package
+            const string_view pathPrefix = pkg.pathPrefixes()[0];
+
+            fmt::format_to(std::back_inserter(buf), "\n{}.pbal_register_package({}, '{}', '{}')\n",
+                           alCfg.registryModule, fullName, mungedName, std::move(pathPrefix));
```

Comment:
Are we doing anything with the munged name?

Reply:
Nope, this is a holdover from the prior `register_package` API. We can get rid of it.